org.eclipse.dawnsci.hdf5.model.test/src/org/eclipse/dawnsci/hdf5/model/internal/HierarchicalDataFileModelTest.java

```diff
@@ -42,46 +48,51 @@ public IHierarchicalDataFile getReader() throws Exception {
 	 */
 	@Test
 	public void printDataContents() throws Exception {
-		try (IHierarchicalDataFile reader = get_i05_4859_Reader.getReader()) {
+		try (NexusFile reader = get_i05_4859_Reader.getReader()) {
 			printDataContents(reader);
 		}
 
 	}
 
-	private void printDataContents(IHierarchicalDataFile reader)
-			throws Exception {
-		Map<String, Object> attributeValues = reader.getAttributeValues();
-		for (Entry<String, Object> entry : attributeValues.entrySet()) {
-			String ln = entry.getKey() + "=" + entry.getValue() + "=="
-					+ HierarchicalDataUtils.extractScalar(entry.getValue());
+	private void printDataContents(NexusFile reader) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)